src/engine/QMLBinding.js

```diff
@@ -24,19 +25,68 @@ class QMLBinding {
     };
   }
 
+  eval(object, context, basePath) {
+    if (this.impl) {
+      // This is needed for `this` support
```

Comment:
perhaps need to better explain that this comment is related to `.call()` instead of direct `.impl(…)`.